components/raftstore/src/store/peer.rs

```diff
@@ -1638,23 +1641,19 @@ where
         };
 
         let mut replicated_idx = self.raft_group.raft.raft_log.persisted;
+        let mut count = 0;
         for (peer_id, p) in self.raft_group.raft.prs().iter() {
-            let store_id = self
-                .region()
-                .get_peers()
-                .iter()
-                .find(|p| p.get_id() == *peer_id)
-                .unwrap()
-                .get_store_id();
-            if !expected_alive_voter.contains(&store_id) {
+            if !expected_alive_voter.contains(peer_id) {
                 continue;
             }
+            count += 1;
             if replicated_idx > p.matched {
                 replicated_idx = p.matched;
             }
         }
+        assert_eq!(count, expected_alive_voter.len());
```

Comment:
Though if PD fails to wait for apply, it indeed has a potential assert fail. Will change it to exit and print a log instead of assertion

Reply:
What I mean is it executes a plan sent by PD and changes its configuration; or have a pending configuration change in uncommitted logs before becoming leader.

Reply:
1. The plan only removes the failed peers, the assertion keeps true.
2. It's a problem. Seems we should record the failed store id instead of alive peers to handle the case conf change is applied during that.